src/v/datalake/record_multiplexer.cc

```diff
@@ -94,17 +94,19 @@ record_multiplexer::record_multiplexer(
 
 ss::future<> record_multiplexer::multiplex(
   model::record_batch_reader reader,
+  kafka::offset start_offset,
```

Comment:
I vaguely remember seeing a similar behavior in some existing tests in some rare cases, this bug was caught by the interruption tests.

Reply:
Nice catch